File: docs/snippets/all/annotation_context_rects.rs
//! Log rectangles with different colors and labels using annotation context

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        rerun::RecordingStreamBuilder::new("rerun_example_annotation_context_rects").spawn()?;

    // Log an annotation context to assign a label and color to each class
    rec.log_static(
        "/",
        &rerun::AnnotationContext::new([
            (1, "red", rerun::Rgba32::from_rgb(255, 0, 0)),
            (2, "green", rerun::Rgba32::from_rgb(0, 255, 0)),
        ]),
    )?;

    // Log a batch of 2 rectangles with different class IDs
    rec.log(
        "detections",
        &rerun::Boxes2D::from_mins_and_sizes([(-2., -2.), (0., 0.)], [(3., 3.), (2., 2.)])
            .with_class_ids([1, 2]),
    )?;

    // Log an extra rect to set the view bounds
    rec.log("bounds", &rerun::Boxes2D::from_half_sizes([(2.5, 2.5)]))?;

    Ok(())
}

File: docs/snippets/all/annotation_context_rects.py
import rerun as rr
import rerun.blueprint as rrb

rr.init("rerun_example_annotation_context_rects", spawn=True)

# Log an annotation context to assign a label and color to each class
rr.log("/", rr.AnnotationContext([(1, "red", (255, 0, 0)), (2, "green", (0, 255, 0))]), static=True)

# Log a batch of 2 rectangles with different `class_ids`
rr.log("detections", rr.Boxes2D(mins=[[-2, -2], [0, 0]], sizes=[[3, 3], [2, 2]], class_ids=[1, 2]))

# Create a Spatial3D View
blueprint = rrb.Blueprint(
    rrb.Spatial3DView(
        origin="/points",
        background=[80, 80, 80],
    )
)

rr.send_blueprint(rrb.Spatial2DView(visual_bounds=rrb.VisualBounds(min=[-4.5, -2.5], max=[2.5, 2.5])))
